components/script/dom/htmlinputelement.rs

```diff
@@ -2549,13 +2580,13 @@ impl VirtualMethods for HTMLInputElement {
         self.upcast::<Element>()
             .check_ancestors_disabled_state_for_form_control();
 
-        for r in radio_group_iter(self, self.radio_group_name().as_ref()) {
-            r.validity_state()
-                .perform_validation_and_update(ValidationFlags::all(), can_gc);
-        }
+        update_related_validity_states(self, can_gc);
     }
 
     fn unbind_from_tree(&self, context: &UnbindContext, can_gc: CanGc) {
+        let form = self.form_owner();
+        let group = self.radio_group_name();
```

Comment:
We should not need to store the group name here. We can call it later without any issues.